docs/superpowers/specs/2026-05-04-rightsizing-advisor-design.md

```
@@ -0,0 +1,301 @@
+- `internal/model/types.go` — add `Rightsizing` + `ContainerRec` + `ResourceRec` value types.
+- `internal/app/messages.go` — add `rightsizingLoadedMsg{key, data, err, generation}`.
+- `internal/app/app_types.go` — add `overlayRightsizing` to the overlay enum + `rightsizingData *k8s.Rightsizing`, `rightsizingLoading bool`, `rightsizingErr error`, `rightsizingGen int` fields.
+- `internal/app/app.go` — `rightsizingCache map[string]*k8s.Rightsizing` field, reset on context/namespace switch.
+- `internal/app/update.go` — wire `rightsizingLoadedMsg` (one switch case).
+- `internal/app/update_actions.go` — append `{Name: "Right-sizing", …}` to the action lists for Pod/Deployment/STS/DS/Job/CronJob; selecting it sets `m.overlay = overlayRightsizing` and dispatches `loadRightsizing`.
+- `internal/app/view_overlays.go` — render the new overlay branch.
+- `internal/app/overlay_hintbar.go` — hint bar entries (`y copy`, `r refresh`, `esc close`, `loading…` when fetching).
+
+## Data flow
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)